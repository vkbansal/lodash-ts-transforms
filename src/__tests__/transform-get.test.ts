import * as ts from 'typescript';
import * as fs from 'fs';
import * as path from 'path';
import transformer from '../transform-get';

const printer = ts.createPrinter();

function getSourceFile(content: string): ts.SourceFile {
    return ts.createSourceFile('filename.ts', content, ts.ScriptTarget.Latest);
}

function expectSnapshot(filename: string) {
    const content = fs.readFileSync(path.resolve(__dirname, 'get', filename), 'utf8');
    const sourceFile = getSourceFile(content);
    const source = printer.printFile(sourceFile);
    const transformedFile = ts.transform(sourceFile, [transformer]).transformed[0];
    const transformed = printer.printFile(transformedFile).trim();

    const snapshot = {
        type: 'transform-baseline',
        filename,
        content,
        source,
        transformed
    };

    expect(snapshot).toMatchSnapshot();
}

const files = fs.readdirSync(path.resolve(__dirname, 'get'), 'utf8');

describe('transform-get tests', () => {
    files.forEach((filename) => {
        test(filename, () => expectSnapshot(filename));
    });
});
