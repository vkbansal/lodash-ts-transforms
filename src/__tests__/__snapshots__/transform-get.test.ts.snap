// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transform-get tests array-default-import.ts 1`] = `
File: array-default-import.ts
##################### Source code #####################
  import get from "lodash/get";
  
  const foo = {};
  const baz = 'somekey';
  
  const a = get(foo, ['a', 'b', 'c', 'd', 'e']);
  const b = get(foo, ['a', '0', 'c', '5', 'e']);
  const c = get(foo, ['a', 0, 'c', 5, 'e']);
  const d = get(foo, \`a[0].c.[\${baz}].e\`);
  
  export { a, b, c, d };
  

############# TypeScript before transform #############
  import get from "lodash/get";
  const foo = {};
  const baz = "somekey";
  const a = get(foo, ["a", "b", "c", "d", "e"]);
  const b = get(foo, ["a", "0", "c", "5", "e"]);
  const c = get(foo, ["a", 0, "c", 5, "e"]);
  const d = get(foo, \`a[0].c.[\${baz}].e\`);
  export { a, b, c, d };
  

############# TypeScript after transform ##############
  import get from "lodash/get";
  const foo = {};
  const baz = "somekey";
  const a = foo && foo.a && foo.a.b && foo.a.b.c && foo.a.b.c.d && foo.a.b.c.d.e;
  const b = foo && foo.a && foo.a[0] && foo.a[0].c && foo.a[0].c[5] && foo.a[0].c[5].e;
  const c = foo && foo.a && foo.a[0] && foo.a[0].c && foo.a[0].c[5] && foo.a[0].c[5].e;
  const d = get(foo, \`a[0].c.[\${baz}].e\`);
  export { a, b, c, d };
`;

exports[`transform-get tests array-named-import.ts 1`] = `
File: array-named-import.ts
##################### Source code #####################
  import { get } from "lodash";
  
  const foo = {};
  const baz = 'somekey';
  
  const a = get(foo, ['a', 'b', 'c', 'd', 'e']);
  const b = get(foo, ['a', '0', 'c', '5', 'e']);
  const c = get(foo, ['a', 0, 'c', 5, 'e']);
  const d = get(foo, \`a[0].c.[\${baz}].e\`);
  
  export { a, b, c, d };
  

############# TypeScript before transform #############
  import { get } from "lodash";
  const foo = {};
  const baz = "somekey";
  const a = get(foo, ["a", "b", "c", "d", "e"]);
  const b = get(foo, ["a", "0", "c", "5", "e"]);
  const c = get(foo, ["a", 0, "c", 5, "e"]);
  const d = get(foo, \`a[0].c.[\${baz}].e\`);
  export { a, b, c, d };
  

############# TypeScript after transform ##############
  import { get } from "lodash";
  const foo = {};
  const baz = "somekey";
  const a = foo && foo.a && foo.a.b && foo.a.b.c && foo.a.b.c.d && foo.a.b.c.d.e;
  const b = foo && foo.a && foo.a[0] && foo.a[0].c && foo.a[0].c[5] && foo.a[0].c[5].e;
  const c = foo && foo.a && foo.a[0] && foo.a[0].c && foo.a[0].c[5] && foo.a[0].c[5].e;
  const d = get(foo, \`a[0].c.[\${baz}].e\`);
  export { a, b, c, d };
`;

exports[`transform-get tests string-default-import.ts 1`] = `
File: string-default-import.ts
##################### Source code #####################
  import get from "lodash/get";
  
  const foo = {};
  const baz = 'somekey';
  
  const a = get(foo, 'a.b.c.d.e');
  const b = get(foo, 'a[0].c[5].e');
  const c = get(foo, \`a[0].c.[\${baz}].e\`);
  
  export { a, b, c };
  

############# TypeScript before transform #############
  import get from "lodash/get";
  const foo = {};
  const baz = "somekey";
  const a = get(foo, "a.b.c.d.e");
  const b = get(foo, "a[0].c[5].e");
  const c = get(foo, \`a[0].c.[\${baz}].e\`);
  export { a, b, c };
  

############# TypeScript after transform ##############
  import get from "lodash/get";
  const foo = {};
  const baz = "somekey";
  const a = foo && foo.a && foo.a.b && foo.a.b.c && foo.a.b.c.d && foo.a.b.c.d.e;
  const b = foo && foo.a && foo.a[0] && foo.a[0].c && foo.a[0].c[5] && foo.a[0].c[5].e;
  const c = get(foo, \`a[0].c.[\${baz}].e\`);
  export { a, b, c };
`;

exports[`transform-get tests string-named-import.ts 1`] = `
File: string-named-import.ts
##################### Source code #####################
  import { get } from "lodash";
  
  const foo = {};
  const baz = 'somekey';
  
  const a = get(foo, 'a.b.c.d.e');
  const b = get(foo, 'a[0].c[5].e');
  const c = get(foo, \`a[0].c.[\${baz}].e\`);
  
  export { a, b, c };
  

############# TypeScript before transform #############
  import { get } from "lodash";
  const foo = {};
  const baz = "somekey";
  const a = get(foo, "a.b.c.d.e");
  const b = get(foo, "a[0].c[5].e");
  const c = get(foo, \`a[0].c.[\${baz}].e\`);
  export { a, b, c };
  

############# TypeScript after transform ##############
  import { get } from "lodash";
  const foo = {};
  const baz = "somekey";
  const a = foo && foo.a && foo.a.b && foo.a.b.c && foo.a.b.c.d && foo.a.b.c.d.e;
  const b = foo && foo.a && foo.a[0] && foo.a[0].c && foo.a[0].c[5] && foo.a[0].c[5].e;
  const c = get(foo, \`a[0].c.[\${baz}].e\`);
  export { a, b, c };
`;
